tensorstore/internal/digest/sha256.h

```diff
@@ -40,6 +45,20 @@ class SHA256Digester {
     return digest;
   }
 
+  std::string HexDigest(bool upper=true) {
```

Comment:
See [absl/strings/escaping.h  [absl::BytesToHexString]](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/escaping.h#L164)

Reply:
This doesn't really belong as a method of SHA256Digester.  But unfortunately absl::BytesToHexString only produces lowercase output.

Reply:
> See [absl/strings/escaping.h [absl::BytesToHexString]](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/escaping.h#L164)

Thanks for mentioning this.

> But unfortunately absl::BytesToHexString only produces lowercase output.

Fortunately, the AWS4 Signature calculations actually only require lower-case input, so will re-use `absl::BytesToHexString` here.